test/test_autograd.py

```diff
@@ -1790,8 +1791,14 @@ class dont_convert(tuple):
     ('addcdiv', (S, S), (0.5, (S, 1), (1, S)), 'scale_broadcast_rhs'),
     ('addcdiv', (1,), (0.5, (S, S, 1), (1, S)), 'scale_broadcast_all'),
     ('zero_', (S, S, S), ()),
-    ('norm', (S, S, S), (2,)),
-    ('norm', (S, S, S), (3,), '3'),
+    ('norm', (S, S), (2,)),
+    ('norm', (S, S), (0,), '0'),
+    ('norm', (S, S), (0.5,), '0_5'),
+    ('norm', (S, S), (1,), '1'),
+    ('norm', (S, S), (3,), '3'),
+    ('norm', (S, S), (-1,), 'neg_1'),
+    ('norm', (S, S), (-0.5,), 'neg_0_5'),
+    ('norm', (S, S), (-1.5,), 'neg_1_5'),
```

Comment:
What? Does `p`-norm on values less or equal 0 even make sense? IIRC it's not even a norm when `p < 1` (no triangle inequality) and it seems that when it's negative it actually puts similar vectors far, and different ones close.

Reply:
As a general rule, if we support something on `Tensor`, we need to make sure the backwards operation computes the derivative correctly.

Reply:
It's still minimized when the vector is small. For a two element vector with p=-2:

![image](https://user-images.githubusercontent.com/655866/32417228-3276d250-c224-11e7-8a6b-21c878a742b1.png) 

[formula](https://www.wolframalpha.com/input/?i=(%7Cx%7C%5E-2%2B%7Cy%7C%5E-2)%5E-0.5)